src/poliastro/core/events.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+from numba import njit as jit
+from numpy.linalg import norm
+
+from poliastro.core.elements import rv2coe
+
+
+@jit
+def continuous_shadow_function_for_penumbra(k, u_, r_sec, R_sec, R_primary):
```

Comment:
Let us name this `eclipse_shadow` so we do not confuse it with the current `shadow_function` from `perturbations.py`. Because the only difference between the `umbra` and `penumbra` logic is a sign, let us use an extra boolean parameter:

```python
def eclipse_function(k, u_, r_sec, R_sec, R_primary, umbra=True):
    # Plus or minus condition
    pm = 1 if umbra is True else -1
    # The eclipse shadow function becomes
    ... + pm * (2 * p * R_primary * cos_psi) ...
```

Therefore, we do not repeat ourselves by having two functions 👍🏽

Reply:
Thanks!